packages/test/src/Test.ts

```diff
@@ -1,30 +1,48 @@
-const colors = require("colors");
-const chai = require("chai");
-const path = require("path");
-const {
-  Web3Shim,
-  createInterfaceAdapter
-} = require("@truffle/interface-adapter");
-const Config = require("@truffle/config");
-const WorkflowCompile = require("@truffle/workflow-compile").default;
-const { Resolver } = require("@truffle/resolver");
-const TestRunner = require("./TestRunner");
-const SolidityTest = require("./SolidityTest");
-const RangeUtils = require("@truffle/compile-solidity/dist/compilerSupplier/rangeUtils");
-const expect = require("@truffle/expect");
-const Migrate = require("@truffle/migrate");
-const { Profiler } = require("@truffle/compile-solidity/dist/profiler");
-const originalrequire = require("original-require");
-const Codec = require("@truffle/codec");
-const debug = require("debug")("lib:test");
-const Debugger = require("@truffle/debugger");
-
-let Mocha; // Late init with "mocha" or "mocha-parallel-tests"
-
-chai.use(require("./assertions"));
-
-const Test = {
-  run: async function (options) {
+import colors from "colors";
+import chai from "chai";
+import path = require("path");
+import { Web3Shim, createInterfaceAdapter } from "@truffle/interface-adapter";
+import Config from "@truffle/config";
+import WorkflowCompile from "@truffle/workflow-compile";
+import { Resolver } from "@truffle/resolver";
+import { TestRunner } from "./TestRunner";
+import { SolidityTest } from "./SolidityTest";
+import RangeUtils from "@truffle/compile-solidity/dist/compilerSupplier/rangeUtils";
+import * as expect from "@truffle/expect";
+import Migrate from "@truffle/migrate";
+import { Profiler } from "@truffle/compile-solidity/dist/profiler";
+import originalrequire from "original-require";
+import * as Codec from "@truffle/codec";
+import debugModule from "debug";
+const debug = debugModule("lib:test");
+import Debugger from "@truffle/debugger";
+import type { Compilation, CompiledContract } from "@truffle/compile-common";
+import Web3 from "web3";
+
+let Mocha: any; // Late init with "mocha" or "mocha-parallel-tests"
+
+chai.use(require("./assertions").default);
+
+type GenerateDebug = (options: {
```

Comment:
The debugger provides a function that is, from what I understand, used for in-test debugging. These files used to import `core/lib/debug/mocha.js`. Exposing this file from core did not seem like a smart idea. This function also needs to be generated using values that are only available here. So in core I generated a function used to generate the `debug` function and pass it to the testing framework so that it can be called here. Yeah, a comment seems good.

Reply:
Maybe `CreateInTestDebugContext`, though a comment would clarify it for me.

Reply:
`createInTestDebugFunction`?

Reply:
Sounds good :)

Reply:
ok so renamed!